pkg/cloud-provider/cloudapi/azure/azure_crd_helpers.go

```diff
@@ -118,13 +168,44 @@ func ComputeVpcToInternalVpcObject(vnet *armnetwork.VirtualNetwork, accountNames
 		}
 	}
 	cidrs := make([]string, 0)
+	var uid string
 	properties := vnet.Properties
-	if properties != nil && properties.AddressSpace != nil && len(properties.AddressSpace.AddressPrefixes) > 0 {
-		for _, cidr := range vnet.Properties.AddressSpace.AddressPrefixes {
-			cidrs = append(cidrs, *cidr)
+	if properties != nil {
+		if properties.AddressSpace != nil && len(properties.AddressSpace.AddressPrefixes) > 0 {
+			for _, cidr := range vnet.Properties.AddressSpace.AddressPrefixes {
+				cidrs = append(cidrs, *cidr)
+			}
+		}
+		if properties.ResourceGUID != nil {
+			uid = strings.ToLower(*properties.ResourceGUID)
 		}
 	}
 
-	return utils.GenerateInternalVpcObject(crdName, accountNamespace, accountName, strings.ToLower(*vnet.Name),
-		strings.ToLower(*vnet.ID), tags, runtimev1alpha1.AzureCloudProvider, region, cidrs, managed)
+	status := &runtimev1alpha1.VpcStatus{
+		CloudName: strings.ToLower(*vnet.Name),
+		CloudId:   strings.ToLower(*vnet.ID),
+		Provider:  runtimev1alpha1.AzureCloudProvider,
+		Region:    region,
+		Tags:      tags,
+		Cidrs:     cidrs,
+		Managed:   managed,
+	}
+
+	labels := map[string]string{
+		config.LabelCloudAccountNamespace: accountNamespace,
+		config.LabelCloudAccountName:      accountName,
+		config.LabelCloudRegion:           region,
+		config.LabelCloudVpcUID:           uid,
+	}
+
+	vpc := &runtimev1alpha1.Vpc{
```

Comment:
vpcObj to be consistent.. for vm you can use vmObj as well

Reply:
Done